history.go

```diff
@@ -8,31 +8,31 @@ import (
 	"log"
 	"os"
 	"os/exec"
+	"os/signal"
 	"strings"
 )
 
 // Recorder store object data for the package
 type Recorder struct {
 	Ctx        context.Context
 	File       io.WriteCloser
-	Path       string
-	Permission os.FileMode
-	Signals    []os.Signal
+	path       string
+	permission os.FileMode
 	Stdin      io.Reader
 	Stdout     io.Writer
 	Stop       context.CancelFunc
 }
 
 // NewRecorder instantiate a new Recorder object and returns a pointer to it.
 func NewRecorder() (*Recorder, error) {
+	ctx, stop := signal.NotifyContext(context.Background(), os.Interrupt)
 	return &Recorder{
-		Ctx:        context.Background(),
-		Path:       "history.log",
-		Permission: 0664,
-		Signals:    []os.Signal{os.Interrupt},
+		Ctx:        ctx,
```

Comment:
Maybe `Context` is clearer here?

Reply:
I think I tried but Context was conflicting with the context.Context. I'll try it again. Maybe the error was somewhere else.

Reply:
updated